examples/nlp/gpt2_text_generation_with_kerasnlp.py

```diff
@@ -0,0 +1,363 @@
+way LLMs can generate coherent text based on user prompts. For a more
+pedagogical discussion on language models, you can refer to the
+[Stanford CS324 LLM class](https://stanford-cs324.github.io/winter2022/lectures/introduction/).
+"""
+
+"""
+## Introduction to KerasNLP
+
+Large Language Models are complex to build and expensive to train from scratch.
+Luckily there are pretrained LLMs available for use right away. One toolkit that
+offers state-of-the-art pretrained models for free is [KerasNLP](https://keras.io/keras_nlp/).
+
+KerasNLP is a natural language processing library that supports users through
+their entire development cycle. KerasNLP offers both pretrained models and
+modularized building blocks, so developers could easily reuse pretrained models
+or stack their own LLM.
+
+In a nutshell, for generative LLM, KerasNLP offers:
+
+- Pretrained models with `generate()` method, e.g.,
+    `keras_nlp.models.GPT2CausalLM` and `keras_nlp.models.OPTCausalLM`.
+- Sampler class that implements generation algorithms such as Top-K, Beam and
+    contrastive search. These samplers can be used to generate text with
+    custom models.
+"""
+
+"""
+## Load a pre-trained GPT-2 model and generate some text
+
+KerasNLP provides a number of pre-trained models, such as [Google
+Bert](https://ai.googleblog.com/2018/11/open-sourcing-bert-state-of-art-pre.html)
+and [GPT-2](https://openai.com/research/better-language-models). You can see
+the list of models available in the [KerasNLP repository](https://github.com/keras-team/keras-nlp/tree/master/keras_nlp/models).
+
+It's very easy to load the GPT-2 model as you can see below:
+"""
+
+# To speed up, we use preprocessor of length 128 instead of full length 1024.
+preprocessor = keras_nlp.models.GPT2CausalLMPreprocessor.from_preset(
+    "gpt2_base_en",
+    sequence_length=128,
+    add_end_token=True,
+)
+gpt2_lm = keras_nlp.models.GPT2CausalLM.from_preset(
+    "gpt2_base_en", preprocessor=preprocessor
+)
+
+"""
+Once the model is loaded, you can use it to generate some text right away. Run
+the cells below to give it a try. It's as simple as calling a single function
+*generate()*:
+"""
+
+start = time.time()
+
+output = gpt2_lm.generate("My trip to Yosemite was", max_length=200)
+print("\nGPT-2 output:")
+print(output)
+
+end = time.time()
+print(f"TOTAL TIME ELAPSED: {end - start:.2f}s")
+
+"""
+Try another one:
+"""
+
+start = time.time()
+
+output = gpt2_lm.generate("That Italian restaurant is", max_length=200)
+print("\nGPT-2 output:")
+print(output)
+
+end = time.time()
+print(f"TOTAL TIME ELAPSED: {end - start:.2f}s")
+
+"""
+Notice how much faster the second call is. This is because the computational
+graph is compiled in the 1st run and re-used in the 2nd behind the scene.
+
+The quality of the generated text looks OK, but we can improved it via
+finetuning.
+"""
+
+"""
+## More on the GPT-2 model from KerasNLP
+
+While it may be sufficient to move on to the next step of finetuning the
```

Comment:
This is a slightly weird intro paragraph flow wise. Maybe just say.

Next up, we will actually fine-tune the model to update it's parameters, but before we do, let's take a look at the full set of tools we have to for working with for GPT2.